### PR TITLE
Added modules for rekey

### PR DESCRIPTION
**Pull request by bchannak**
Vault supports rekey in order to change the number of key-shares, key-thresholds, PGP keys, etc. of a running Vault instance. 

Added the following modules for rekey:
- hashivault_rekey_init: Start new rekey
- hashivault_rekey: Update rekey (by providing current unseal keys)
- hashivault_rekey_cancel: Cancel a rekey in progress
- hashivault_rekey_status: Get status/progress of the on-going rekey

### CODE REVIEW

File: ansible/modules/hashivault/hashivault_rekey.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+DOCUMENTATION = '''
+---
+module: hashivault_rekey
+version_added: "1.3.2"
```
Comment (TerryHowe) [inline ansible/modules/hashivault/hashivault_rekey.py]: s/1.3.2/3.3.0

File: ansible/modules/hashivault/hashivault_rekey_cancel.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+DOCUMENTATION = '''
+---
+module: hashivault_rekey_cancel
+version_added: "1.3.2"
```
Comment (TerryHowe) [inline ansible/modules/hashivault/hashivault_rekey_cancel.py]: s/1.3.2/3.3.0

File: ansible/modules/hashivault/hashivault_rekey_init.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+DOCUMENTATION = '''
+---
+module: hashivault_rekey_init
+version_added: "1.3.2"
```
Comment (TerryHowe) [inline ansible/modules/hashivault/hashivault_rekey_init.py]: s/1.3.2/3.3.0

File: ansible/modules/hashivault/hashivault_rekey.py (hunk 1 above)
Comment (bchannak) [inline ansible/modules/hashivault/hashivault_rekey.py]: Updated version_added. I will work on adding functional tests for rekey.

**Comment by TerryHowe** [comment]
Thanks!

**Pull request merged**